### PR TITLE
.github: convert issue templates to issue forms

### PR DESCRIPTION
**Pull request by benesch**
I threw up a preview of what the new issue forms will look like here: https://github.com/benesch/mz-issue-templates/issues/new/choose

----

Issue forms are a new GitHub feature that replace issue templates. They
render the desired questions with proper form widgets (textareas,
dropdowns, etc.), which makes for a *much* nicer experience when filing
issues.

This commit converts our non-internal issue templates to these new issue
forms.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/03-documentation.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    attributes:
+      label: Feature description
+  - type: input
+    attributes:
+      label: Engineering tracking ticket
+  - type: textarea
+    attributes:
+      label: Design documents
+      description: |
+        Please link any available design documents, like technical specs, product specs, or test files.
```
Comment (morsapaes) [inline .github/ISSUE_TEMPLATE/03-documentation.yml:30]:
```suggestion
  - type: textarea
    attributes:
      label: What pages are affected?
  - type: textarea
    attributes:
      label: What needs to be changed or added?
  - type: input
    attributes:
      label: Engineering tracking ticket
   - type: textarea
    attributes:
      label: Feature description
  - type: textarea
    attributes:
      label: Design documents
      description: |
        Please link any available design documents, like technical specs, product specs, or test files.
```

Comment (morsapaes) [reply .github/ISSUE_TEMPLATE/03-documentation.yml:30]: Minor, but it's more intuitive to me to have the issue first and everything else afterwards.

**Review by morsapaes — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/03-documentation.yml (hunk 1 above)
Comment (benesch) [reply .github/ISSUE_TEMPLATE/03-documentation.yml:30]: Sounds good to me! I figured you might want to do some deeper surgery on these forms, and particular this one, so I just stuck with a pretty rote translation of the existing templates.

**Comment by andrioni** [comment]
This is really neat!

**Pull request merged**